DOCS-9680 Commerce Cloud Shop Connector (#1184)
mulesoft/docs-connectors: commerce-cloud-shop-api/1.0/modules/ROOT/pages/shop-api-connector-troubleshoot.adoc

= Commerce Cloud B2C Shop Connector - Troubleshoot - Mule 4

These are some common troubleshooting suggestions for Anypoint Connector for Salesforce Commerce Cloud B2C Shop APIs (B2C Shop Connector).

*400 Bad Request*

A `400 Bad Request` error response can occur if:

> * The URI parameters such as Organization Id, Site Id and Version parameter not passed with the proper value.

Reviewer: ```suggestion
* The values of URI parameters such as `Organization Id`, `Site Id`, and `Version` are incorrect.
```

* The above parameter passed with wrong value

> * The above parameter passed with an extra space

Reviewer: ```suggestion
*  The values of URI parameters such as `Organization Id`, `Site Id`, and `Version` include an extra space.
```

== Solution

> Correct and pass the proper values.

Reviewer: ```suggestion
Correct the values.
```

*401 Unauthorized*

A `401 Unauthorized` error response can occur if:

* The client ID used in the configuration is incorrect.
* The client secret used in the configuration is incorrect.

== Solution

> 

Reviewer: GLOBALLY reword to avoid "pass."

> * Correct and pass the proper values for the client ID and client secret.

Reviewer: ```suggestion
* Correct the client ID and client secret values.
```

> * If the Client ID and client secret are both correct and you are still getting the error, contact the Commerce Cloud team.

Reviewer: ```suggestion
* If the client ID and client secret are both correct and you still get the error, contact the Commerce Cloud team.
```

*403 Forbidden*

A `403 Forbidden` error response can occur if the scopes used in the configuration are incorrect.

== Solution

> Correct and pass the proper value for the scopes.

Reviewer: ```suggestion
Correct the scope values.
```

*405 Method Not Allowed*

> A `405 Method Not Allowed` error can occur if a resource that previously supported HTTP POST got removed and had its HTTP GET method deprecated.

Reviewer: ```suggestion
A `405 Method Not Allowed` error can occur if a resource that previously supported HTTP POST no longer does and its HTTP GET method becomes deprecated.
```
Are these two separate conditions or one condition as a result of the other? Clarify.

== Solution

Contact the Commerce Cloud team.

*415 Unsupported Media Type*

A `415 unsupported media type` error response can occur if:

> * The body type the user is passing to the connector `Content` parameter does not match the required input type.

Reviewer: ```suggestion
* The body type of the `Content` parameter does not match the required input type.
```

> * Expression mode is enabled and the output type of the body is not specified as `application/json`.

Reviewer: ```suggestion
* Expression mode is enabled and the output type of the body is not `application/json`.
```

== Solution

* Use the correct output type for the body.

> * Enable Expression mode and specify the output type of the body as `application/json`.

Reviewer: ```suggestion
* If Expression mode is enabled, specify the output type of the body as `application/json`.
```
